### DISCUSSION TITLE
Possible to live on the edge?

### DISCUSSION THREAD

**Discussion by AlexAllocated** [LazyVim/LazyVim#3625] [General]
A few times now I've noticed that on `main` there will be a commit that fixes an issue I'm having, but it has yet to make it into a `stable` release. I know these are usually released pretty fast, but still, I'm weird lol. I'm someone who likes living on the edge with this stuff and I'm curious if there's a simple way to do that?

I've tried changing the `"LazyVim/LazyVim"` spec to specify the `main` branch and set `version` to `false`:

```lua
require("lazy").setup({
    spec = {
        -- Import LazyVim built-in specs.
        { "LazyVim/LazyVim", branch = "main", import = "lazyvim.plugins", version = false },
        -- Import my own specs.
        { import = "plugins" }
    }
    -- truncated for post snippet
})
```

But I noticed that the `lazyvim.plugins` import [returns specs](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/init.lua#L13-L16) with `version="*"` anyway. I can then include those specs manually and override them:

```lua
require("lazy").setup({
spec = {
    -- Import LazyVim plugins module from latest on main.
    { "LazyVim/LazyVim", branch = "main", import = "lazyvim.plugins", version = false },
    -- Override returned specs with latest on main.
    { "folke/lazy.nvim", branch = "main", version = false },
    { "LazyVim/LazyVim", branch = "main", cond = true, lazy = false, opts = {}, priority = 10000, version = false },
    -- Import my own specs.
    { import = "plugins" },
},
```

It appears to work but I think there is some double-loading of some kind going on and causing some weird side effects so I reverted this change for now.

Curious if there is a simpler way to live on the edge 😅

**Comment by Old-Farmer** [comment]
See <https://www.lazyvim.org/configuration/lazy.nvim>

**Reply by AlexAllocated** [reply]
Oh wow good catch! Looks like I was already doing it mostly right then. I've pored over those docs many times and still missed that bit. ~~I assume I still need to leave the one doing `import = "lazyvim.plugins"` though right? Essentially I'm overriding [the specs returned by that one](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/init.lua#L13-L16) I assume.~~ Considering they are adding those specs in a `plugins/core.lua` file I think it's implied you have to leave the `import="lazyvim.plugins"` spec in place.

Thanks for the link!

**Reply by Old-Farmer** [reply]
Put spec where you can override the default LazyVim plugin spec. Adding sth in the first import spec ` { "LazyVim/LazyVim", import = "lazyvim.plugins" }` can't work because LazyVim plugin spec will override it(import is always later than adding the current plugin spec). So A file in lua/plugins is a nice place. Putting a spec just after the first import spec can work, but is not a prefered way for structured configuration.
